common/src/main/java/org/apache/uniffle/common/compression/Codec.java

```diff
@@ -29,6 +29,8 @@ public abstract class Codec {
   public static Codec newInstance(RssConf rssConf) {
     Type type = rssConf.get(COMPRESSION_TYPE);
     switch (type) {
+      case NONE:
+        return null;
```

Comment:
It could encounter NPE if return null, maybe you can return a `NoneCodec` which will do nothing but return `src` while call `compress(byte[] src)`.

The following test code will encounter NPE.

```java
compressData = Codec.newInstance(new RssConf()).compress(data);
```

Reply:
`Codec.decompress` requires a new buffer, which is not necessary for no compression. Can we make `Codec#newInstance` return `Optional<Codec>`?

Reply:
I see. It can be better to return `Optional<Codec>` I think.

Reply:
Although the NONE type looks weired for me, but in this case, I will +1 for this to bypass codec for gluten